### PR TITLE
[feature] Export data to CSV

### PR DESCRIPTION
**Pull request by xbill82**
You waited long for it, it is here. This PR allows to export items currently shown in a table view as a CSV file.
Ok, this is a first iteration, it might need more varnishing to make everyone happy.

Kapture 2021-11-05 at 11 58 26

## Other improvements

* Column view needed some serious love
* MOAR tests on Column view

### CODE REVIEW

**Comment by jenow** [comment]
That's a great addition but I don't think it suits most of users needs.
If we can't export the whole collection that's very limitating. I know the feature has just been added to this existing view with pagination but it would be better to maybe have a specific view for this which can actually be usable with date pickers to select a time range. Adding features is great but they have to actually be useful.
In the meantime I think adding a checkbox "export whole collection" and date pickers would make this feature make more sense from a user point of view

**Comment by xbill82** [comment]
@jenow yes, your statement makes sense. We are currently thinking about the next iterations on this feature and will take your ideas into account.

**Review by rolljee — APPROVED** [review]

File: src/components/Data/Documents/Views/Column.vue [hunk 1]
```diff
@@ -313,6 +324,39 @@ export default {
     }
   },
   methods: {
+    isObject: _.isObject,
+    promptExportCSV() {
+      this.$bvModal
```
Comment (Njuelle) [inline src/components/Data/Documents/Views/Column.vue:329]: Nitpicking: you can `await` the confirmation modal instead of attaching a `then` callback

**Review by Njuelle — APPROVED** [review]
I tried a massive export, the file is OK (however it is all messed up in "LibreOffice" idk why).

But I think we'll have to give the column view a boost before we can release this feature.
When I select a lot of fields to display in the table (from 20 fields), the admin console becomes very slow or even unusable

**Comment by xbill82** [comment]
@Njuelle I actually took a look at the problem and it seems very costly to solve this problem. With 500 rows, the page becomes very bloated, as you state (regardless of the number of selected fields). My feeling is that the bottleneck is the bootstrap table component. Do you think the ROI of blocking this feature while we dig into that component (or something else) is interesting? Shouldn't we rather release this and let the feedback come to us?

**Comment by Njuelle** [comment]
@xbill82 Indeed, let's make a release of this feature (at least on next-console) and deal with the perf problem on large dataset in a second time

**Pull request merged**